packages/profile-sync-controller/src/controllers/user-storage/UserStorageController.ts

```diff
@@ -473,6 +473,53 @@ export default class UserStorageController extends BaseController<
     );
   }
 
+  #isLocked = false;
+
+  readonly #pendingOperations: (() => Promise<unknown> | unknown)[] = [];
+
+  /**
+   * Locks the controller, preventing operations that are subsequently called with `executeWithLockHandling` from executing.
+   * These operations will be queued until the controller is unlocked.
+   */
+  lock() {
```

Comment:
Nice!!

Who is responsible for locking the controller? Is it handled internally, or something else will lock/unlock it?

Reply:
OK saw the work in extension, really cool!